### ISSUE TITLE
Hunter Huts and how AIs use them

### ISSUE THREAD

**Issue by Panbutt** [Sh0wdown/UnofficialCrusaderPatch#168]
Hunter Huts are an interesting food building and sadly one that is rarely, if not ever used. With the AI, they're rather bugged. You can place them down manually in AIVs however doing so means the AI will even build them on maps with no deer. As a result, hunting huts can take up much needed population as the AIs will never put them to sleep.

Is it possible to change how AIs handle Hunting Huts? The subject was brought up before a while ago, but I think it warrants a little more discussion as hunting huts can improve the economy of AIs.

My train of thought was to use Woodcutter AI logic for Hunting Huts. For Woodcutters, the AI will place a set amount and eventually demolish them the less trees are on the map. I think this should be applied to Huts as well. AIs will build a set amount, but eventually put them to sleep or demolish them as deer populations dwindle.

I'm unsure if this is already in the game, but it would also be greatly beneficial if deer repopulated as time went on. A condition could be in place where when the deer population hits a near-extinction level, the AI will put all Hunting Huts to sleep and allow the deer to repopulate to a certain percentage. I'm unsure how much finagling this would require.

If this does not seem feasible, I would at least like it if AIs put Huts to sleep when there are no deer. Instead of having them use up peasants to just stand there.

**Comment by Lolasik011** [comment]
Hunting hut ! #46 I also suggested a topic but I was ignored.
I would also like to see what happens.

**Comment by Lolasik011** [comment]
it's my fault that I closed the topic

**Comment by Panbutt** [comment]
> It would indeed be nice, but I think it's rather complex, so don't put your hopes up.

Understandable, it sounds like you'd have to wrestle with multiple things. Like said, at least allowing AIs to put huts to sleep when there's no deer is a good starting point.

**Comment by Heroesflorian** [comment]
Also, one thing I always found a bit sad is that hunters shoot hostile soldiers in SH1 and in SHC scenarios, but not on SHC skirmish maps... I always really liked the "military potential" of hunters in SH1. For custom scenarios they work really well, and for SHC skirmish-scenarios they'd be great, too. 

I could very well imagine a few hunters e.g. in/around the Sultan's castles to strengthen his defenses by helping pick off hostile troops that approach when there is no deer on the map, or go hunting and collect meat when there is deer. Certainly, the hunters would stand no chance to withstand an army, but they would make it less viable to snipe off Sultan's slingers and fire throwers with a handful of archers.

**Comment by PitchNeeded** [comment]
Yes if woodcutters can attack enemies then definitively hunters should be able to I think!

**Comment by PitchNeeded** [comment]
This is unlikely to be able to be implemented, but I would love for the water on Stronghold Crusader maps to actually be used, and make fish a resource the hunters in hunters' huts can collect if there is no deer :D

Obviously it wouldn't be as productive as hunting deer, as water is more common on maps than deer. The animation would be hunters entering water, disappearing for a while (to simulate diving under the water for fish) and emerging with the fish (which would just be the meat they usually carry). They would then carry it to the granary.

**Comment by PitchNeeded** [comment]
Also, perhaps rabbits and camels on the map should be targeted by hunters huts, not just deer.
